documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,301 @@
+They should not be coupled to allow evolving API independent of the storage layer.
+This is the reason why the API is using the `FruitDto`.
+This class will be used by the quarkus to convert JSON to java object for client request and java object to JSON for the response.
+[source,java]
+----
+public class FruitDto {
+
+  private String name;
+  private String description;
+
+  public FruitDto() {}
+
+  public FruitDto(String name, String description) {
+    this.name = name;
+    this.description = description;
+  }
+  // getters and setters omitted for brevity
+}
+----
+
+One important thing to note is that having a default constructor is required by the JSON serialization layer.
+
+== Configuring the Cassandra database
+The main two properties to configure are `contact-points` to access to Cassandra and `local-datacenter` to use when connecting.
+
+A sample configuration should look like this:
+
+[source,properties]
+----
+# configure the Cassandra client for a replica set of two nodes
+quarkus.cassandra.contact-points={cassandra_ip}:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter={dc_name}
+----
+
+In this example, we are using a single instance running on localhost:
+
+[source,properties]
+----
+# configure the Cassandra client for a single instance on localhost
+quarkus.cassandra.contact-points=127.0.0.1:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter=datacenter1
+----
+
+You can configure all java-driver settings using the `application.conf` or `application.json` files.
+They will be passed automatically to the underlying java-driver config loader.
+The settings from `application.properties` with `quarkus.cassandra` prefix have the priority.
+
+To see the full list of settings, please refer to the {reference-conf-url}[reference.conf]
+
+== Running a Cassandra Database
+As by default, `CassandraClient` is configured to access a local Cassandra database on port 9042 (the default Cassandra port).
+If you have a local running database on this port, you should assert that:
+`quarkus.cassandra.load-balancing-policy.local-datacenter` setting is equal to dc of your local cassandra instance.
+
+If you want to use Docker to run a Cassandra database, you can use the following command to launch one:
+[source,shell]
+----
+docker run \
+   --name local-cassandra-instance \
+   -p 7000:7000 \
+   -p 7001:7001 \
+   -p 7199:7199 \
+   -p 9042:9042 \
+   -p 9160:9160 \
+   -p 9404:9404 \
+   -d \
+   launcher.gcr.io/google/cassandra3
+----
+
+Next, you need to create the keyspace:
+[source,shell]
+----
+docker exec -it local-cassandra-instance cqlsh -e "CREATE KEYSPACE IF NOT EXISTS k1 WITH replication = {'class':'SimpleStrategy', 'replication_factor':1}"
+----
+
+And the Cassandra table that will be used by the quickstart application.
+[source,shell]
+----
+docker exec -it local-cassandra-instance cqlsh -e "CREATE TABLE IF NOT EXISTS k1.fruit(id text, name text, description text, PRIMARY KEY((id), name))"
+----
```

Comment:
I wonder if it might read more cleanly to combine these into a single paragraph and show both code samples after the fact:

"Next you need to create the keyspace and table that will be used by your application:"
<sample1>
<sample2>

Reply:
It is split now already:
```

Next, you need to create the keyspace:
[source,shell]
----
docker exec -it local-cassandra-instance cqlsh -e "CREATE KEYSPACE IF NOT EXISTS k1 WITH replication = {'class':'SimpleStrategy', 'replication_factor':1}"
----

And the Cassandra table that will be used by the quickstart application.
[source,shell]
----
docker exec -it local-cassandra-instance cqlsh -e "CREATE TABLE IF NOT EXISTS k1.fruit(id text, name text, description text, PRIMARY KEY((id), name))"
```

Reply:
He was asking the opposite: that you combine the snippets together. I think I agree it would read more cleanly.